docs/syntax_and_semantics/autocasting.md

```diff
@@ -0,0 +1,66 @@
+# Type autocasting
+
+At function calls, Crystal transparently casts elements of certain types when there is no ambiguity.
```

Comment:
Oh, I was wrong there. Autocasting is not limited to function calls. It also applies to assignments. Maybe I missed something else?
```suggestion
Crystal transparently casts elements of certain types when there is no ambiguity.
```

Reply:
I added an example for the other situation where this works.